### PR TITLE
Add configuration to pairwise subject for jwt tokens

### PR DESCRIPTION
**Pull request by RivinduM**
### Proposed changes in this pull request

This pr add configuration to enable pairwise subject identifier for jwt tokens.
```
[oauth.oidc.subject_type]
default_subject_type="public"
enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection=true
```

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 1]
```diff
@@ -839,6 +839,9 @@
                 {% endif %}
             </FAPI>
             {% endif %}
+            {% if oauth.oidc.enable_pairwise_subject_identifier_for_access_token is defined %}
```
Comment (RivinduM) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:842]: add default config for subject type

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -839,6 +839,14 @@
                 {% endif %}
             </FAPI>
             {% endif %}
+            {% if oauth.oidc.subject_type.default_subject_type is defined %}
+                <DefaultSubjectType>{{oauth.oidc.subject_type.default_subject_type}}</DefaultSubjectType>
+            {% else %}
+                <DefaultSubjectType>public</DefaultSubjectType>
+            {% endif %}
+            {% if oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_on_pairwise_subject_type_selection is defined %}
+                <EnablePairwiseSubForAccessToken>{{oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_on_pairwise_subject_type_selection}}</EnablePairwiseSubForAccessToken>
```
Comment (RivinduM) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:848]:
```suggestion
                <EnablePairwiseSubForAccessToken>{{oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection}}</EnablePairwiseSubForAccessToken>
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 3]
```diff
@@ -839,6 +839,14 @@
...
+            {% endif %}
```
Comment (RivinduM) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:849]:
```suggestion
            {% if oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection is defined %}
                <EnablePairwiseSubForAccessToken>{{oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection}}</EnablePairwiseSubForAccessToken>
            {% endif %}
```

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 4]
```diff
@@ -887,6 +887,14 @@
                     {% endfor %}
                 </AllowedSignatureAlgorithms>
             </FAPI>
+            {% if oauth.oidc.subject_type.default_subject_type is defined %}
+                <DefaultSubjectType>{{oauth.oidc.subject_type.default_subject_type}}</DefaultSubjectType>
+            {% else %}
+                <DefaultSubjectType>public</DefaultSubjectType>
+            {% endif %}
```
Comment (janakamarasena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:894]: Lets add this via the default.json then we can remove the conditional check.

**Comment by janakamarasena** [comment]
Lets add a sample deployment.toml config to the PR description.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 4 above)
Comment (RivinduM) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: fixed with [f2bd155](https://github.com/wso2/carbon-identity-framework/pull/4908/commits/f2bd15540efd6caff6fd5220da54ea527d2ad59d)

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 5]
```diff
@@ -887,6 +887,12 @@
                     {% endfor %}
                 </AllowedSignatureAlgorithms>
             </FAPI>
+            {% if oauth.oidc.subject_type.default_subject_type is defined %}
```
[the remark this answers is not in the document]
Comment (RivinduM) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: removed with [301d02c](https://github.com/wso2/carbon-identity-framework/pull/4908/commits/301d02c85900e5835cff1b2779b7737f9872d947)

**Review by janakamarasena — DISMISSED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 6]
```diff
@@ -242,6 +242,8 @@
   "oauth.oidc.fapi.allowed_signature_algorithms": ["PS256", "ES256"],
 
   "oauth.oidc.fapi.enable_validation": true,
+  "oauth.oidc.subject_type.default_subject_type": "public",
+  "oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection": true
```
Comment (janakamarasena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:246]:
```suggestion
  "oauth.oidc.subject_type.enable_pairwise_subject_for_access_tokens_if_pairwise_subject_type_selection": true,
```

**Review by janakamarasena — APPROVED** [review]

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**